lib/app.dart

```diff
@@ -79,17 +219,38 @@ class SailAppState extends State<SailApp> with WidgetsBindingObserver {
     await settings.setValue(ThemeSetting(newValue: themeToLoad));
   }
 
+  bool get inInitialSetup => theme == null || !binariesStarted;
+
+  Widget loadingIndicator() {
+    if (!binariesStarted) {
+      return SailColumn(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        mainAxisAlignment: MainAxisAlignment.center,
+        spacing: SailStyleValues.padding25,
+        children: [
+          SailText.primary20('Mainchain starting up: $mainchainStartupMessage'),
+          SailText.primary20('Sidechain starting up: $sidechainStartupMessage'),
+          const SailCircularProgressIndicator(),
+        ],
+      );
+    }
+    return const SailCircularProgressIndicator();
+  }
+
   @override
   Widget build(BuildContext context) {
-    if (theme == null) {
+    if (inInitialSetup) {
       return MaterialApp(
         home: Material(
           color: _sidechain.chain.color,
           child: SailTheme(
             data: SailThemeData.lightTheme(_sidechain.chain.color),
-            child: const SailScaffold(
+            child: SailScaffold(
               body: Center(
-                child: SailCircularProgressIndicator(),
+                // TODO: better error message with troubleshooting tips here
+                child: _initBinariesError != null
+                    ? SailText.primary24('Ran into unrecoverable error on startup: $_initBinariesError')
+                    : loadingIndicator(),
```

Comment:
@octobocto input appreciated. When this branch is triggered we're at a unrecoverable state, and most likely it's the programmer's fault. Shouldn't happen with the end user, but if it happens they should get a good error message